### ISSUE TITLE
Responses $refs empty (no example available)

### ISSUE THREAD

**Issue by shubaivanqbee** [nelmio/NelmioApiDocBundle#1859]
I use
            "name": "nelmio/api-doc-bundle",
            "version": "v4.3.0",
            
with  "symfony/framework-bundle": "5.3.3",            
            
And execute evrything like in documentation, added new schema and try to used in by ref

config/packages/nelmio_api_doc.yaml

```
nelmio_api_doc:
    documentation:
        components:          
            schemas:
                NelmioImageList:
                    description: "Response for some queries"
                    type: object
                    properties:
                        total:
                            type: integer
                            example: 42
```                            

and my action

```
     * @OA\Response(
     *     response=200,
     *     description="List of image definitions",
     *     @OA\JsonContent(@OA\Schema(
     *       type="object",
     *       title="ListOperationsResponse",
     *       additionalProperties={"$ref": "#/components/schemas/NelmioImageList"}
     *     ))
     * )
     * 
```

but I faced with no example available, could someone help?  

I executed 

php bin/console a:install
and totally cleared cache
rm -rf var/cache/ && php bin/console c:c && chmod 777 -R var/cache/

the same case when I use commom Operation annotation

    use Nelmio\ApiDocBundle\Annotation\Operation;
    use OpenApi\Annotations as OA;

     * @Operation(
     *     tags={"Stripe"},
     *     summary="test",
     *     @OA\RequestBody(
     *         request="body",
     *         description="test"
     *     ),
     *     @OA\Response(
     *         response=200,
     *         description="test",
     *         @OA\JsonContent(@OA\Schema(
     *              type="object",
     *              title="test",
     *              additionalProperties={"$ref": "#/components/schemas/NelmioImageList"}
     *        ))
     *     )
     * )
     
   

As you can see in screen in global Schema section models viewed correct, perhaps some problem with `ref` or ..

![image](https://user-images.githubusercontent.com/78418206/128864299-752a7351-cfb7-4ee4-b996-ee40556475e4.png)

the same case when I used

```
     * @Operation(
     *     tags={"Stripe"},
     *     summary="test",
     *     @OA\RequestBody(
     *         request="body",
     *         description="test"
     *     ),
     *     @OA\Response(
     *         response=200,
     *         description="test",
     *         @OA\JsonContent(@OA\Schema(
     *              type="object",
     *              example={"foo": "bar", "hello": "world"}
     *         ))
     *     )
     * )
```

looks like global problem with response model, but everything works correct and show corerct when I used Model annotation

```
     * @Operation(
     *     tags={"Stripe"},
     *     summary="test",
     *     @OA\RequestBody(
     *         request="body",
     *         description="test",
     *         required=true,
     *         @Model(
     *              type=MyRequest::class,
     *              groups={MyRequest::API_DOC}
     *         )
     *     ),
     *     @OA\Response(
     *         response=200,
     *         description="test",
     *         @Model(
     *              type=MyResponse::class,
     *              groups={MyResponse:API_DOC}
     *         )
     *     )
     * )
     * 
```

I updated to v4.4.0 but not helped, waht I'm doing wrong ?

**Comment by jgauthi** [comment]
Hello,
I had the same problem and fix it with this changes:

before: `@OA\JsonContent(@OA\Schema(`
after: `@OA\JsonContent(`

You must remove after the `)` too much. Can you try ?

**Comment by shubaivanqbee** [comment]
@jgauthi Wow, several days I fighted with that problem, thanks

```
     * @Operation(
     *     tags={"Stripe"},
     *     summary="create-customer-portal",
     *     @OA\RequestBody(
     *         request="body",
     *         description="create-customer-portal"
     *     ),
     *     @OA\Response(
     *         response=200,
     *         description="Stripe billingPortal session",
     *         @OA\JsonContent(
     *              type="object",
     *              title="Stripe billingPortal session",
     *              additionalProperties={"$ref": "#/components/schemas/StripeSession"}
     *        )
     *     )
     * )
```
![Выделение_051](https://user-images.githubusercontent.com/78418206/129166243-6e30a950-1b52-4acb-9923-93724f022b44.png)

and when I changed it to 
```

     * @Operation(
     *     tags={"Stripe"},
     *     summary="create-customer-portal",
     *     @OA\RequestBody(
     *         request="body",
     *         description="create-customer-portal"
     *     ),
     *     @OA\Response(
     *         response=200,
     *         description="Stripe billingPortal session",
     *         @OA\JsonContent(
     *              type="object",
     *              title="Stripe billingPortal session",
     *              ref="#/components/schemas/StripeSession"
     *        )
     *     )
     * )
```

I got expected result, one object with properties

![Выделение_052](https://user-images.githubusercontent.com/78418206/129166594-2061b78f-08d4-47fe-ad07-ea1875c20e42.png)

**Comment by GuilhemN** [comment]
Indeed, references must be directly defined at the schema level in zircote/swagger-php.

Did some documentation mislead you into using `additionalProperties` instead? It would be great if we could improve this for newcomers :)

**Comment by macr1408** [comment]
> Indeed, references must be directly defined at the schema level in zircote/swagger-php.
> 
> Did some documentation mislead you into using `additionalProperties` instead? It would be great if we could improve this for newcomers :)

That's literally the example that's currently in the documentation https://symfony.com/bundles/NelmioApiDocBundle/current/faq.html

I spent many hours dealing with this issue because of this! @GuilhemN the bundle definitely needs a better documentation, at the very least adding more and complete examples would be really helpful

**Comment by GuilhemN** [comment]
Oh indeed, thanks for pointing that out @macr1408! It's a quite old example, from the development of v4 and I guess it must have passed our review during a big diff, and somehow nobody mentioned it until now.

I proposed a fix in https://github.com/nelmio/NelmioApiDocBundle/pull/2004. I'm not completely satisfied even by the structure of the docs, but I unfortunately now can only spend little time on this bundle, and I prioritize fixing bugs and answering issues.
Contributions are always very welcome though :)

**Comment by macr1408** [comment]
> Oh indeed, thanks for pointing that out @macr1408! It's a quite old example, from the development of v4 and I guess it must have passed our review during a big diff, and somehow nobody mentioned it until now.
> 
> I proposed a fix in #2004. I'm not completely satisfied even by the structure of the docs, but I unfortunately now can only spend little time on this bundle, and I prioritize fixing bugs and answering issues. Contributions are always very welcome though :)

Thanks I appreciate it, I know that supporting these open source projects can consume a lot of your free time.

I might create a pull request later if I find something good to add, thanks again!
